utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -69,34 +70,40 @@ enum class ArchiveType(vararg val extensions: String) {
 }
 
 /**
- * Unpack the [File] to [targetDirectory].
+ * Unpack the [File] to [targetDirectory] using [filter] to select only the entries of interest.
  */
-fun File.unpack(targetDirectory: File) =
+fun File.unpack(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) =
     when (ArchiveType.getType(name)) {
-        ArchiveType.SEVENZIP -> unpack7Zip(targetDirectory)
-        ArchiveType.ZIP -> unpackZip(targetDirectory)
+        ArchiveType.SEVENZIP -> unpack7Zip(targetDirectory, filter)
+        ArchiveType.ZIP -> unpackZip(targetDirectory, filter)
 
-        ArchiveType.TAR -> inputStream().unpackTar(targetDirectory)
-        ArchiveType.TAR_BZIP2 -> BZip2CompressorInputStream(inputStream()).unpackTar(targetDirectory)
-        ArchiveType.TAR_GZIP -> GzipCompressorInputStream(inputStream()).unpackTar(targetDirectory)
-        ArchiveType.TAR_XZ -> XZCompressorInputStream(inputStream()).unpackTar(targetDirectory)
+        ArchiveType.TAR -> inputStream().unpackTar(targetDirectory, filter)
+        ArchiveType.TAR_BZIP2 -> BZip2CompressorInputStream(inputStream()).unpackTar(targetDirectory, filter)
+        ArchiveType.TAR_GZIP -> GzipCompressorInputStream(inputStream()).unpackTar(targetDirectory, filter)
+        ArchiveType.TAR_XZ -> XZCompressorInputStream(inputStream()).unpackTar(targetDirectory, filter)
 
-        ArchiveType.DEB -> unpackDeb(targetDirectory)
+        ArchiveType.DEB -> unpackDeb(targetDirectory, filter)
 
         ArchiveType.NONE -> {
             throw IOException("Unable to guess compression scheme from file name '$name'.")
         }
     }
 
 /**
- * Unpack the [File] assuming it is a 7-Zip archive. This implementation ignores empty directories and symbolic links.
+ * Unpack the [File] assuming it is a 7-Zip archive. This implementation ignores empty directories and symbolic links
+ * and all entries not matched by the given [filter].
  */
-fun File.unpack7Zip(targetDirectory: File) {
+fun File.unpack7Zip(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) {
+    val skip = skipFilter(
```

Comment:
This code is really confusing. The first parameter is called `skip` but you name the full object also `skip` locally. Would it be not better to name the local variable `filter` ?
Same question everywhere.

Reply:
I think the name `skip` is correct in this context, as the resulting function returns *true* when an entry should be skipped. This becomes obvious in the `if (skip(entry))` statement. Maybe the name of the function `skipFilter()` is a bit misleading.

Reply:
I understand. But nevertheless when following the call chain, it gets confusing.

You can leave it as if if you want.